### ISSUE TITLE
Connexion issue

### ISSUE THREAD

**Issue by Jackseed** [blockframes/blockframes#4837]
We have a good number of users that reported a connexion issue. The workflow is always the same but the bug doesn't appear every time.
Steps to reproduce:
1. Create an org through classic sign up flow
2. Invite a new user to that org through CRM as blockframes admin (`/c/o/admin/panel/organization/orgId`)
3. Try to connect as if you were that new user with the pre-generated password received by email

Bug appears step 3, users receive a snackbar message "The password is invalid or the user does not have a password".

**Comment by SamirTouira** [comment]
Finally, after many tries, impossible to reproduce this. However, when I copy/paste the pregenerated password with the space at the start in the credentials email, I got the same error than these users. I thought about one thing, simply my question is : did you asked them if they have correctly copied the pregenerated password? By removing the space from the beginning of the password?

![image](https://user-images.githubusercontent.com/52285231/106791036-b41cbb80-6654-11eb-859a-818784d5b292.png)

**Comment by Jackseed** [comment]
Ok thanks @SamirTouira, good idea for the space, maybe we can add something to remove spaces on password... @GrandSchtroumpf ?
Cause on the email when you double click on the password it doesn't take the space anyway

**Comment by GrandSchtroumpf** [comment]
This is a little bit dangerous if the user actually has a space at the beginning or end of his password. For my part I've got a bunch of spaces in my pwd, but not at the beginning or end. @phpgeekfr do you know if it safe to trim a password ?

**Comment by phpgeekfr** [comment]
@GrandSchtroumpf Yes it is safe to trim() the input password. same for the login input we can trim

**Comment by SamirTouira** [comment]
Alright, the trim solution works. Thank you for your answers

**Comment by Jackseed** [comment]
I reopen this issue because I'm a bit worried about changing password handling that fast. 
@phpgeekfr @GrandSchtroumpf will this be retro-compatible with old passwords that have white spaces in it?

**Comment by phpgeekfr** [comment]
@Jackseed erf you are right .. It is indeed possible to have a spaces in password. I did'nt knew this was allowed by google

I created an account with password `     1` and it worked .

**Comment by phpgeekfr** [comment]
so maybe we need to revert ? Now that we understand the problem we could tell users to not copy with first space from email ?

**Comment by phpgeekfr** [comment]
But maybe we should prevent new users to create a password containing spaces ? Unfortunately we cannot know if there is some existing users in this situation..

**Comment by Jackseed** [comment]
> so maybe we need to revert ? Now that we understand the problem we could tell users to not copy with first space from email ?

Yes unfortunately seems to be the only possible action at the moment. 

We could prevent whitespaces from now but since there will be old users with it (or at least we have to assume that) it means handling 2 use cases instead of one. Looks too complicated to me, @GrandSchtroumpf ?

**Comment by phpgeekfr** [comment]
> We could prevent whitespaces from now but since there will be old users with it (or at least we have to assume that) it means > handling 2 use cases instead of one. Looks too complicated to me, @GrandSchtroumpf ?

**Comment by phpgeekfr** [comment]
so @SamirTouira I think you should remove trim() for login and also update the signup form to prevent users to put spaces in the input.
@GrandSchtroumpf Do you think this is a good idea?

**Comment by SamirTouira** [comment]
I'm sorry, I have closed the issue too fast. Yes i will do, indeed it's more complicated than a simple trim(). I think @phpgeekfr has a good idea about this issue

**Comment by GrandSchtroumpf** [comment]
I'm not sure to solve the issue describe above when you copy the space for the mail. For me there is no reason to prevent user to add a space in the password. The real solution would be to not put a space between after the ":" in the mail.

**Comment by phpgeekfr** [comment]
> I'm not sure to solve the issue describe above when you copy the space for the mail. For me there is no reason to prevent user to add a space in the password. The real solution would be to not put a space between after the ":" in the mail.

That should work too !

**Comment by Jackseed** [comment]
> I'm not sure to solve the issue describe above when you copy the space for the mail. For me there is no reason to prevent user to add a space in the password. The real solution would be to not put a space between after the ":" in the mail.

Honestly I think it would be even worse as then if you double click on the password you get everything. 
```
Example
password:lalalala
```

I guess there is no technical solution anyway so lets remove that trim() and I close that issue. 

Marie is warned for your advice @SamirTouira and she will advise next customers with this problem to mind white spaces. We will also try to keep that case alive if it happens again so we may have some concrete tests on this

**Comment by GrandSchtroumpf** [comment]
If you double click on a text with ":" at the beginning, the ":" is not selected. So in your exemple "password:blablabla" if you double click on "blabla" you won't get the ":". Plus if ever it select everything the user will notice the overline on the "password" I think ;)

**Comment by Jackseed** [comment]
I re-open that issue since Marie had another user facing this problem. She tried for the user to enter the password (without space) and still faced the issue. 
She had to debug that user by deleting it and recreating it but at least we know it's not a space question (or not always). 

I put it on hold waiting for a use case we can work on but if you have any idea on what could cause this please shoot here as it's an important one. 
@GrandSchtroumpf @phpgeekfr

**Comment by Jackseed** [comment]
Marie is suggesting we contact Google for that question..?

**Comment by SamirTouira** [comment]
In parallel I can continue some tests to creating new accounts until I trigger this issue

**Comment by Jackseed** [comment]
I don't want to waste too much time in this with blind testing I think testing is good but if we have some strategy for specific use cases we assume could cause the problem

**Comment by phpgeekfr** [comment]
@GrandSchtroumpf @SamirTouira Here is a PR https://github.com/blockframes/blockframes/pull/4895 to allow more verbosity in password creation.
This will help us to gather more information when the case occurs

**Comment by phpgeekfr** [comment]
@Jackseed  Is the problem occurring only when user is created via CRM or also when email is invited (toJoinoOrg or AttendEvent)  or both ?

**Comment by Jackseed** [comment]
For now the flow always has been the one described in description. It doesn't mean that it couldn't appear otherwise but most of our users are created this way:

> Steps to reproduce:
> 
> 1. Create an org through classic sign up flow
> 2. Invite a new user to that org through CRM as blockframes admin (`/c/o/admin/panel/organization/orgId`)
> 3. Try to connect as if you were that new user with the pre-generated password received by email
> 
> Bug appears step 3, users receive a snackbar message "The password is invalid or the user does not have a password".

**Comment by Jackseed** [comment]
I close this now that the auth flow has been updated, let's see if that case happens again
